lib/modules/navigationMiddleware.js

```diff
@@ -6,16 +6,30 @@ export function matchRoute(path, routes) {
   let route;
 
   for (route of routes) {
-    const [url, handler] = route;
+    const [url, handler, meta] = route;
     const reg = pathToRegex(url);
     const result = reg.exec(path);
 
     if (result) {
-      return { handler, reg, result };
+      return { handler, reg, result, meta };
     }
   }
 }
 
+const getRouteMeta = (routes, shouldSetPage, data) => {
+  const { method, pathName } = data;
+  const route = matchRoute(pathName, routes);
+
+  if (route && route.meta) {
+    const { meta } = route;
+
+    if (shouldSetPage && method === METHODS.GET) {
+      return meta;
+    }
+  }
+  return undefined;
```

Comment:
right, that's the point. It's the same as saying, "we meant to set this to to `null`". `undefined` occurs naturally and implicitly in cases where variables are incorrectly handled and therefore can have overloaded semantics.